### ISSUE TITLE
Can't use `match_id - 1` as a list index, even though `match_id` is 1-based

### ISSUE THREAD

**Issue by keenancrane** [penrose/penrose#1536]
Ok, here's a really annoying bug that I don't know what to do about:
- For reasons I don't remember, we use 1-based indexing for `match_id`
- This means that if you collect matches into a list `foo`, you may want to access elements of that list later via `foo[ match_id - 1 ]`
- But (gotcha!) `match_id - 1` isn't a valid index into a list, because it's some kind of autodiff type instead of float :cry: 
- So, I get a compiler error `Invalid indexing`. The error gets triggered from `compiler/Style.ts` on the marked line below.

![image](https://github.com/penrose/penrose/assets/3604525/d84cf57b-49bc-4ed8-89b9-c570b7a4610d)

For instance, this pattern doesn't work:
```
collect Point p into ps
where In(c,p)
foreach Curve c {
   xs = listof x from ps
   c.points = xs
}

forall Point p; Curve c
where In(c,p) {
   -- error triggered here
   vec2 x = ( c.points[match_id - 1][0], c.points[match_id - 1][1] )
}
```

Note that there's of course a simpler solution here (just access `p.x`), but in the bigger context I'm working (simplifying one of @jiriminarcik's programs), that solution won't work.

TL;DR: Is there a small change we can make here so that the expression `match_id - 1` can be used as an index?
